_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,344 @@
+By default, Flink manages resources in coarse-grained units called *slots*, which are slices
+of a TaskManager's resources. By default, Streaming pipelines fill a slot with one parallel
+subtask of each operator, so each slot holds a pipeline of subtasks.
+Through *'slot sharing groups'*, users can influence how subtasks are assigned to slots.
+
+With fine-grained resource management, TaskManager slots can now be dynamically sized.
+Transformations and Operators specify what resource profiles they would like (size of CPU,
+memory pools, disk space) and Flink's Resource Manager and TaskManagers slice off that specific
+part of a TaskManagers total resources. Think of it as a minimal lightweight resource orchestration
+layer within Flink. The figure below illustrates the difference between the existing default way (shared
+fixed-size slots) and the new fine-grained resource management.
+
+<figure style="align-content: center">
+  <img src="{{ site.baseurl }}/img/blog/2021-09-25-release-1.14.0/fine_grained_resource_management.png" style="display: block; margin-left: auto; margin-right: auto; width: 600px"/>
+</figure>
+
+Hmmmm, why would we even do that, is this not better handled directly by Kubernetes or Yarn?
+For many cases, that is in fact true. There are, however, some reasons to not rely purely on K8s/Yarn for this:
+
+  - For many small slots, the overhead of dedicated TMs is very high (JVM overhead, Flink control data structures).
+    Slot-sharing works around this implicitly by sharing the slots between all operator types, which means
+    sharing resources between lightweight (need small slots) and heavyweight (need large slots) operators.
+    However, this works only well when all operators share the same parallelism, which isn't aways optiomal.
+    Furthermore, sometimes certain operators work better when run in isolation (for example ML training operators
+    that need dedicated GPU resources).
+  - K8s and Yarn take often quite some time to fulfill requests, especially on loaded clusters.
+    For certain batch jobs, a of efficiency gets lost waiting for Yarn and K8s to fullfill the requests.
+    
+Who should use this feature? For most jobs (both streaming and batch) the default resource management mechanism
+are perfectly fine. This feature can help you increase the resource efficiency, if you have either long-running
+streaming jobs, or fast batch jobs, where different stages have different resource requirements, and you possibly
+already tuned the parallelism of different operators to different values. Then this feature can help you
+increase the resource efficiency.
+Alibaba's internal Flink-based platform has used this mechanism for some time now and has significantly
+increased the resource utilization of the cluster. 
+
+Please refer to the [Fine-grained Resource Management Docs](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/)
+for details on how to use this feature.
```

Comment:
```suggestion
You may be wondering whether this would be better handled directly by Kubernetes or YARN. 
Here are some reasons why you may not want to rely purely on these popular resource managers:

  - For many small slots, the overhead of dedicated TaskManagers is very high (JVM overhead, Flink control data
    structures). Slot-sharing implicitly works around this by sharing the slots between all operator types, which    
    means sharing resources between lightweight operators (which need small slots) and heavyweight operators   
    (which need large slots). However, this only works well when all operators share the same parallelism, which is 
    not always optimal. Furthermore, certain operators sometimes work better when run in isolation (i.e. training  
    operators for machine learning that need dedicated GPU resources).
  - Kubernetes and YARN often take quite some time to fulfill requests, especially on loaded clusters.
    For certain batch jobs, efficiency gets lost while waiting for the requests to be fulfilled. 
    
So when should you use this feature? For most batch and streaming jobs, the default resource management mechanism is perfectly suitable. Fine-grained resourced management can help you increase resource efficiency if you have either long-running streaming jobs or fast batch jobs, where different stages have different resource requirements, and you may have already tuned the parallelism of different operators to different values. 
Alibaba's internal Flink-based platform has used this mechanism for some time now and the resource utilization of the cluster has significantly increased. 

Please refer to the [Fine-grained Resource Management documentation](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/)
for details on how to use this feature.
```